src/validate-icon.c

```diff
@@ -237,24 +267,42 @@ main (int argc, char *argv[])
   g_autoptr(GOptionContext) context = NULL;
   g_autoptr(GError) error = NULL;
 
-  context = g_option_context_new ("PATH");
+  context = g_option_context_new (NULL);
   g_option_context_add_main_entries (context, entries, NULL);
   if (!g_option_context_parse (context, &argc, &argv, &error))
     {
       g_printerr ("Error: %s\n", error->message);
       return 1;
     }
 
-  if (argc != 2)
+  if (opt_path != NULL && opt_fd != -1)
     {
-      g_printerr ("Usage: %s [OPTION…] PATH\n", argv[0]);
+      g_printerr ("Error: Only --path or --fd can be given\n");
+      return 1;
+    }
+
+  if (opt_path)
+    {
+      opt_fd = g_open (opt_path, O_RDONLY | O_CLOEXEC, 0);
```

Comment:
it's closed anyways once the validator terminates, no?

Reply:
Yes, but it's not cleaned up by your code but by the OS eventually which is a leak. So while I'd still prefer a local autofd, this isn't bad.

Reply:
don't we actually also leak `opt_path`?

Reply:
I think it's pretty hard to close the fd in the right cases, since it depends on whether we run in sandbox and whether the fd given to the validator has FD_CLOSEXEC. I would leave it as is.

Reply:
Don't really see the issue... Use a local `autofd` variable for cleaning up the fd the code creates itself and always assign to opt_fd as well. Same for `opt_path` with `autofree`.

Not leaking things is really nice because tools like valgrind and lsan will work better.